### PR TITLE
Fix application delete.

### PR DESCRIPTION
**Pull request by jortel**
Need to delete all associations else FK constraint violation.
Also, needed cascade delete.

### CODE REVIEW

File: api/group.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func (h StakeholderGroupHandler) Delete(ctx *gin.Context) {
 		h.deleteFailed(ctx, result.Error)
 		return
 	}
-	result = h.DB.Select("Stakeholders").Delete(m)
+	result = h.DB.Select(clause.Associations).Delete(m)
```
Comment (jortel) [inline api/group.go:127]: Better syntax to delete all associations.

File: api/stakeholder.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func (h StakeholderHandler) Delete(ctx *gin.Context) {
 		h.deleteFailed(ctx, result.Error)
 		return
 	}
-	result = h.DB.Select("Groups").Delete(m)
+	result = h.DB.Select(clause.Associations).Delete(m)
```
Comment (jortel) [inline api/stakeholder.go:127]: Better syntax to delete all associations.

File: model/application.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ type Application struct {
 	Facts             JSON
 	Comments          string
 	Tasks             []Task     `gorm:"constraint:OnDelete:CASCADE"`
-	Tags              []Tag      `gorm:"many2many:applicationTags"`
+	Tags              []Tag      `gorm:"many2many:applicationTags;constraint:OnDelete:CASCADE"`
```
Comment (jortel) [inline model/application.go:16]: Add DB constraint for completeness.

**Review by ibolton336 — APPROVED** [review]
👍  LGTM

**Pull request merged**